packages/components/src/DataTable/Column/DataTableCell.tsx

```diff
@@ -93,4 +112,7 @@ export const DataTableCell = styled(DataTableCellLayout)`
       text-decoration: underline;
     }
   }
+  :focus {
+    border: 1px solid ${({ theme }) => theme.colors.key};
+  }
```

Comment:
I think the browser default `outline` might be the blue ring that we're seeing. And for the "focus visible" behavior (keyboard vs clicking), take a look at how `Tab.tsx` does it – the border style you have here should only be on when `focusVisible` is true.

(Don't worry about adding focus visible behavior to `DataTableHeaderCell` – it should already be on there from `useClickable`.)

```suggestion
  :focus {
    outline: none;
  }
```